tests/multinode.rs

```diff
@@ -631,94 +631,102 @@ fn test_multi_node_dynamic_network() {
-                    "{:x} {} get_balance: {:?} leader_balance: {}",
-                    server.0.debug_id(),
-                    i,
-                    getbal,
-                    leader_balance
-                );
-                let bal = getbal.unwrap_or(0);
-                let distance = (leader_balance - bal) / 500;
-                max_distance = max(distance, max_distance);
-                total_distance += distance;
-                if distance > max_distance_increase {
-                    info!("Node {:x} is behind by {}", server.0.debug_id(), distance);
-                    max_distance_increase = distance;
-                    if max_distance_increase as u64 > purge_lag as u64 {
-                        server.1.exit();
-                        info!("Node {:x} is exiting", server.0.debug_id());
-                        retain_me = false;
-                    }
-                }
-                if distance > 0 {
-                    num_nodes_behind += 1;
-                }
-                if let Some(bal) = getbal {
-                    if bal == leader_balance {
-                        success += 1;
-                    }
-                }
-                retain_me
-            });
-            if num_nodes_behind != 0 {
-                info!("{} nodes are lagging behind leader", num_nodes_behind);
-            }
-            info!(
-                "SUCCESS[{}] {} out of {} distance: {} max_distance: {}  finality: {}",
-                i,
-                success,
-                validators.len(),
-                total_distance,
-                max_distance,
-                get_finality(&leader_data)
-            );
-            if success == validators.len() && total_distance == 0 {
-                consecutive_success += 1;
-            } else {
-                consecutive_success = 0;
-                failures += 1;
-            }
-            if consecutive_success == 10 {
-                break;
-            }
+
+        last_finality = finality;
+
+        let balance = retry_get_balance(&mut client, &bob_pubkey, Some(expected_balance));
+        assert_eq!(balance, Some(expected_balance));
+        consecutive_success += 1;
+
+        info!(
+            "SUCCESS[{}] balance: {}, finality: {} ms",
+            i, expected_balance, last_finality,
+        );
+
+        if consecutive_success == 10 {
+            break;
         }
     }
+
+    info!("Took {} s to converge", duration_as_s(&start.elapsed()),);
+    info!("Checking balance of validators and calculating distance");
+    //verify all validators have the same balance
+    let mut max_distance = 0i64;
+    let mut total_distance = 0i64;
+    let mut num_nodes_behind = 0i64;
+    validators.retain(|server| {
+        let mut retain_me = true;
+        let mut client = mk_client(&server.0);
+        trace!("{:x} get_balance start", server.0.debug_id());
+        let getbal = retry_get_balance(&mut client, &bob_pubkey, Some(expected_balance));
```

Comment:
i would change this to just confirm the last sig in all the validators instead of retry get balance.